Feature: Support numbered pagination.

llmII opened the pull request:
**EDIT - NOTE:**
Look at further comments, the basis on which we created this PR has changed. This is mostly a historical comment to see how we got to where we are in later comments.

## What does this PR do?
Show page number plus one, and use a Go button instead of a "Next Page" button. This allows the user to jump more than +/-1 pages at a time.

<!-- MANDATORY -->

<!-- explain the changes in your PR, algorithms, design, architecture -->
Just modified some templates and LESS files to make the pagination number visible.

## Why is this change important?

<!-- MANDATORY -->

<!-- explain the motivation behind your PR -->

Allows for jumping pages by number, instead of being limited +/-1. Also gives the user feedback as to what page is next so that they know where they are presently.

Firefox Desktop
![image](https://user-images.githubusercontent.com/40866124/210290900-74e16fb0-c422-4ed1-af92-4720c2903764.png)

Chrome Emulating iPhone XR dimensions
![image](https://user-images.githubusercontent.com/40866124/210291004-5d0e5a5d-0fac-42c2-8b33-f482609da2a9.png)


## How to test this PR locally?

<!-- commands to run the tests or instructions to test the changes-->

``` sh
make themes.all
make run
```

## Author's checklist

<!-- additional notes for reviewiers -->

Might need a  more in depth check for mobile UX related things.

## Related issues
Closes #2067

mrpaulblack reviewed (changes requested):
Hi @llmII ,
Thanks for your PR. We discussed it internally and decided that we like the pagination feature that you are proposing, but not in its current form.

We would either prefer the current impl. with a previous and next page button or a proper pagination widget like for example it is impl in google:
![image](https://user-images.githubusercontent.com/38733246/210423497-56d473f7-d166-492f-938f-ed8194d79766.png)

Please comment back if you want to refactor your PR with the proposed change or not (so we can close it, but I hope you will still contribute to our project with your ideas), thanks :D

Thanks for your great idea tho. Also looking at your commit please check out https://docs.searxng.org/dev/quickstart.html -> We have make commands to make the static files commit -> `make static.build.commit`.

llmII committed:
Feature: Pagination Widget with list of numbers.

Adds to the navigation widget, preserving forward/backward nav, and
inserting a list of clickable page numbers between them.

Phone sized devices continue without this widget as deterministic
display under small screen sizes has not been solved.

The widget is agnostic to the actual amount of pages there are that one
can navigate to and as such shows all plausible, albeit not necessarilly
valid, possibilities.

This widget does not interfere with infinite scroll in any fashion.

mrpaulblack committed:
[fix] use input elements for pagination requests

* use html input elements instead of buttons for the pagination forms at the bottom of the result page
* move the less section that hides the pagination number widget on mobile to the mobile section
* clean up the less code for styling the numbers of the pagination widget
* fix: add the margin for box results (like in news category) to the bottom of the result to have a margin between pagination widget and article result

mrpaulblack committed:
[build] /static

mrpaulblack reviewed (approved):
I finally made the review for this PR. Sorry I took so long for this.

I made some small changes in my review and pushed them in a second commit on top. I also rebased on current master. Hope that was ok.

Really cool feature. I tested this for a while on RTL, LTR, with general, image, news and mixed results on desktop, tablet and phone layout with center enabled as well as center results disabled on light and dark theme.

One small nit I noticed:
on center results set to true the results get smaller then the defined min result width on tablet so it can happen on some edge cases that the number 1 and 10 is cut off by the next and previous page button. But that is IMO ok, since its such an edge case.

Everything else LGTM, thanks a lot, really cool feature :+1:
